### PR TITLE
feat: store key frame indices when loading video tensor from uri

### PR DESCRIPTION
**Pull request by anna-charlotte**
Goals:

When calling `.load_uri_to_video_tensor` we want to additionally store the indices of the key frames in the Document's `.tags['key_frame_indices']`. This makes it easier to access selected frames, e.g.:
```python
from docarray import Document

d = Document(uri='toy.mp4')
d.load_uri_to_video_tensor()
first_scene = d.tensor[d.tags['key_frame_indices'][0]: d.tags['key_frame_indices'][1]]

print(first_scene.shape)
```
```
(95, 320, 176, 3)
``` 
The indices will be stored independently of the value of the `key_frames` parameter.

- [x] store key frame indices in `.tags`
- [x] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: docarray/document/mixins/video.py [hunk 1]
```diff
@@ -110,16 +110,36 @@ def load_uri_to_video_tensor(
...
+            stream = container.streams.video[0]
+            stream.codec_context.skip_frame = 'NONKEY'
 
-            frames = []
+            key_frames = []
             for frame in container.decode(video=0):
+                img = frame.to_image()
+                key_frames.append(img)
+
+        with av.open(self.uri, **kwargs) as container:
```
Comment (samsja) [inline docarray/document/mixins/video.py:121]: if we only want key frame we don't need to repopen the files again

File: docarray/document/mixins/video.py [hunk 2]
```diff
@@ -110,16 +110,36 @@ def load_uri_to_video_tensor(
...
+            i = 0
+            curr_key_frame = key_frames[i]
+            key_frame_indices = []
+            frames = []
+            for j, frame in enumerate(container.decode(video=0)):
                 img = frame.to_image()
                 frames.append(np.asarray(img))
 
-        self.tensor = np.moveaxis(np.stack(frames), 1, 2)
+                if (np.asarray(img) == np.asarray(curr_key_frame)).all():
```
Comment (samsja) [inline docarray/document/mixins/video.py:130]: I am not a big fan of doing this. 
From what I understand av is able to tell when it load the video which frame is a key frame. It means that the info is either store in the file or that av can determine it. In both case we should do smth like this.

```python
for j, frame in enumerate(container.decode(video=0)):
    frames.append(np.asarray(img))
    if is_key_frame(frame):
        key_frame_indices.append(j)

```
the question is how de we do the `is_key_frame` but I believe av must have it

File: docarray/document/mixins/video.py (hunk 1 above)
Comment (samsja) [reply docarray/document/mixins/video.py:125]: No, rather it is when `only_keyframes=true` a part of your current code is not needed it is executed but you return a result that you compute earlier so you can just skip it when you only want the keyframes

**Review by JoanFM — APPROVED** [review]

File: docarray/document/mixins/video.py [hunk 3]
```diff
@@ -100,7 +100,7 @@ def generator_from_webcam(
     def load_uri_to_video_tensor(
         self: 'T', only_keyframes: bool = False, **kwargs
     ) -> 'T':
-        """Convert a :attr:`.uri` to a video ndarray :attr:`.tensor`.
+        """Convert a :attr:`.uri` to a video ndarray :attr:`.tensor` and store the indices of the keyframes in :attr:`.tags` if only_keyframes=False.
```
Comment (JohannesMessner) [inline docarray/document/mixins/video.py:103]: Maybe we can mention the key in tags that is used for this?

File: docs/datatypes/video/index.md [hunk 4]
```diff
@@ -81,6 +81,22 @@ for idx, c in enumerate(d.tensor):
 
 Makes sense, right?
```
Comment (JohannesMessner) [inline docs/datatypes/video/index.md:82]:
```suggestion
```

Not related to this PR, but there are always some people to whom things don't make sense, let's not mock them lol

**Review by samsja — APPROVED** [review]
lgtm

File: docarray/document/mixins/video.py [hunk 5]
```diff
@@ -102,7 +102,8 @@ def load_uri_to_video_tensor(
     ) -> 'T':
         """Convert a :attr:`.uri` to a video ndarray :attr:`.tensor`.
 
-        :param only_keyframes: only keep the keyframes in the video
+        :param only_keyframes: if true keep only the keyframes in the video, if false keep all frames and store the
+            indices of the keyframes in :attr:`.tags`
```
Comment (alexcg1) [inline docarray/document/mixins/video.py:106]:
```suggestion
        :param only_keyframes: if True keep only the keyframes, if False keep all frames and store the
            indices of the keyframes in :attr:`.tags`
```

File: docs/datatypes/video/index.md [hunk 6]
```diff
@@ -79,7 +79,22 @@ for idx, c in enumerate(d.tensor):
 :width: 40%
 ```
 
-Makes sense, right?
+
+Additionally, when `only_keyframes=False` the indices of the keyframes are being stored in `.tags['keyframe_indices']` when calling `.load_uri_to_video_tensor`. This way, one can easily access selected scenes:
```
Comment (alexcg1) [inline docs/datatypes/video/index.md:83]:
```suggestion
Additionally, if `only_keyframes=False` the keyframe indices will be stored in `.tags['keyframe_indices']` when you call `.load_uri_to_video_tensor()`. This way, you can easily access selected scenes: 
```

**Review by alexcg1 — CHANGES_REQUESTED** [review]

**Review by alexcg1 — APPROVED** [review]
LGTM :+1:

**Pull request merged**